components/server/src/common.rs

```diff
@@ -715,7 +717,14 @@ pub fn build_hybrid_engine(
         )
     }
 
-    memory_engine.start_cross_check(disk_engine.clone(), region_cache_engine_context.pd_client());
+    memory_engine.start_cross_check(
+        disk_engine.clone(),
+        region_cache_engine_context.pd_client(),
+        Box::new(|| {
+            let ctx = GC_CONTEXT.lock().unwrap();
```

Comment:
What if compaction_filter is not enable? Seems this `GC_CONTEXT` can only be used when compaction_filter is enabled.

Reply:
Good catch. For simplicity, cross check is only supported when compaction filter is enabled.